sdk/core/azure-core/inc/azure/core/http/body_stream.hpp

```diff
@@ -211,34 +207,84 @@ namespace Azure { namespace Core { namespace Http {
     int64_t OnRead(Azure::Core::Context const& context, uint8_t* buffer, int64_t count) override;
 
   public:
-#if defined(AZ_PLATFORM_POSIX)
     /**
      * @brief Construct from a file.
      *
-     * @param fd File descriptor.
-     * @param offset Offset in the file to start providing the data from.
-     * @param length Length of the data, in bytes, to provide.
+     * @param file A pointer to an already opened file object that can be used to identify the file.
+     *
+     * @remark The caller owns the file object and needs to open it along with keeping it alive for
+     * the necessary duration. The caller is also responsible for closing it once they are done.
      */
-    FileBodyStream(int fd, int64_t offset, int64_t length)
-        : m_fd(fd), m_baseOffset(offset), m_length(length), m_offset(0)
+    FileBodyStream(FILE* file) : m_hFile(file), m_offset(0)
```

Comment:
@JeffreyRichter please review this API design change.

Reply:
I think the offset enables reading one file with multiple bodyStreams in parallel. We should re-consider removing it from ctror.